src/core/trulens/core/tru.py

```diff
@@ -821,6 +829,81 @@ def get_leaderboard(
                 .sort_values(by=feedback_cols, ascending=False)
             )
 
+    def add_ground_truth_nowait(
+        self,
+        ground_truth: mod_groundtruth_schema.GroundTruth,
+    ) -> None:
+        """Add a ground truth object to the queue to be inserted in the next batch."""
+        if self.batch_thread is None:
+            self.batch_thread = threading.Thread(
+                target=self.batch_ground_truth_loop, daemon=True
+            )
+            self.batch_thread.start()
+        self.batch_ground_truth_queue.put(ground_truth)
+
+    def batch_ground_truth_loop(self):
+        while True:
+            time.sleep(self.GROUND_TRUTHS_BATCH_TIMEOUT)
+            ground_truths = []
+            while True:
+                try:
+                    ground_truth = self.batch_ground_truth_queue.get_nowait()
+                    ground_truths.append(ground_truth)
+                except queue.Empty:
+                    break
+            if ground_truths:
+                try:
+                    self.db.batch_insert_ground_truth(ground_truths)
+                except Exception as e:
+                    # Re-queue the ground truth objects that failed to be inserted
+                    for ground_truth in ground_truths:
+                        self.batch_ground_truth_queue.put(ground_truth)
+                    logger.error(
+                        "Re-queued ground truth objects due to insertion error {}",
+                        e,
+                    )
+                    continue
+
+    def add_ground_truth_to_dataset(
+        self,
+        dataset_name: str,
+        ground_truth_df: pd.DataFrame,
+        dataset_metadata: Optional[Dict[str, Any]] = None,
+    ):
+        """Create a new dataset, if not existing, and add ground truth data to it. If
+        the dataset with the same name already exists, the ground truth data will be added to it.
+
+        Args:
+            dataset_name: Name of the dataset.
+            ground_truth_df: DataFrame containing the ground truth data.
+            dataset_metadata: Additional metadata to add to the dataset.
+        """
+
+        # Create and insert the dataset record
+        dataset = mod_dataset_schema.Dataset(
+            name=dataset_name,
+            meta=dataset_metadata,
+        )
+        dataset_id = self.db.insert_dataset(dataset=dataset)
+
+        for _, row in ground_truth_df.iterrows():
+            ground_truth = mod_groundtruth_schema.GroundTruth(
+                dataset_id=dataset_id,
+                query=row["query"],
+                query_id=row.get("query_id", None),
+                expected_response=row.get("expected_response", None),
+                expected_chunks=row.get("expected_chunks", None),
+                meta=row.get("meta", None),
+            )
+            self.add_ground_truth_nowait(ground_truth)
+
+    def get_ground_truth_from_dataset(self, dataset_name: str) -> pd.DataFrame:
```

Comment:
We can simplify the name `get_ground_truth` given we pass dataset_name in params?

Reply:
there exists a `get_ground_truth` method that will fetch a single row of gt data given the id - it's not currently being used but I just add it in case it might be useful in downstream evaluation runs.

Reply:
actually nvm - I misread and thought you were reviewing the DAO / ORM PR. 

Will update.